addons/website/static/src/builder/plugins/options/website_page_config_option_plugin.js

```diff
@@ -134,70 +186,119 @@ export class BaseWebsitePageConfigAction extends BuilderAction {
-        headerEl.style.removeProperty("background-color");
-        headerEl.classList.forEach((cls) => {
-            if (cls.startsWith("bg-o-color-")) {
-                headerEl.classList.remove(cls);
-            }
-        });
+    /**
+     * Remove any background color class or inline style from the given
+     * target element.
+     *
+     * @param {'header' | 'breadcrumb'} type The element type to reset.
+     */
+    resetColor(type) {
+        const el = this.websitePageConfig.getTarget(type);
+        el.style.removeProperty("background-color");
+        const classes = [...el.classList].filter((cls) => cls.startsWith("bg-o-color-"));
+        if (classes.length) {
+            el.classList.remove(...classes);
+        }
     }
 
-    resetTextColor() {
-        const headerEl = this.document.querySelector("#wrapwrap > header");
-        headerEl.style.removeProperty("color");
-        headerEl.classList.forEach((cls) => {
-            if (cls.startsWith("text-o-color-")) {
-                headerEl.classList.remove(cls);
-            }
-        });
+    /**
+     * Remove any text color class or inline style from the given
+     * target element.
+     *
+     * @param {'header' | 'breadcrumb'} type The element type to reset.
+     */
+    resetTextColor(type) {
+        const el = this.websitePageConfig.getTarget(type);
+        el.style.removeProperty("color");
+        const classes = [...el.classList].filter((cls) => cls.startsWith("text-o-color-"));
+        if (classes.length) {
+            el.classList.remove(...classes);
+        }
     }
 }
 export class SetWebsiteHeaderVisibilityAction extends BaseWebsitePageConfigAction {
     static id = "setWebsiteHeaderVisibility";
     apply({ editingElement, value: headerPositionValue, isPreviewing }) {
-        const lastValue = this.websitePageConfig.getVisibilityItem();
+        const lastValue = this.websitePageConfig.getVisibilityItem("header");
         this.history.applyCustomMutation({
-            apply: () => this.visibilityHandlers[headerPositionValue](),
-            revert: () => this.visibilityHandlers[lastValue](),
+            apply: () => this.headerVisibilityHandlers[headerPositionValue](),
+            revert: () => this.headerVisibilityHandlers[lastValue](),
         });
 
+        // update the breadcrumb's top whenever header's visibility changes
+        const breadcrumbEl = this.websitePageConfig.getTarget("breadcrumb");
+        if (breadcrumbEl && editingElement) {
+            breadcrumbEl.style.top = `${editingElement.getBoundingClientRect().height}px`;
+        }
         this.websitePageConfig.setDirty(isPreviewing);
     }
     isApplied({ editingElement, value }) {
-        return this.websitePageConfig.getVisibilityItem() === value;
+        return this.websitePageConfig.getVisibilityItem("header") === value;
+    }
+}
+export class SetWebsiteBreadcrumbVisibilityAction extends BaseWebsitePageConfigAction {
```

Comment:
More or less like ```SetWebsiteHeaderVisibilityAction```, maybe we can have common action like ```SetWebsiteHeaderBreadcrumbVisibilityAction``` and based on some attribute we will check whether action is called for header or breadcrumb.